test/integration/buildclient_test.go

```diff
@@ -233,5 +233,5 @@ func (t *testBuildOpenshift) Close() {
 	defer t.lock.Unlock()
 	close(t.stop)
 	t.server.CloseClientConnections()
-	t.server.Close()
+	defer t.server.Close()
```

Comment:
For some reason without it I still hit race error on `t.server.Close()`, as if `CloseClientConnections` was blocking this. I've tried removing the call `CloseClientConnections` (per documentation only `Close` is required) but that hanged the entire test. Thus I come up with this, I'm not saying it's the best solution, but it works.